File: app/src/main/java/com/smarteval/app/data/model/Alumni.kt
package com.smarteval.app.data.model

data class AlumniRequest(
    val id: String,
    val alumniId: String,
    val alumniUsername: String,
    val alumniEmail: String,
    val professorId: String,
    val professorEmail: String,
    val status: AlumniRequestStatus,
    val createdAt: String
)

enum class AlumniRequestStatus {
    PENDING, APPROVED, REJECTED
}

File: lib/models/alumni.dart
import 'package:json_annotation/json_annotation.dart';

part 'alumni.g.dart';

@JsonSerializable()
class AlumniRequest {
  final String id;
  final String alumniId;
  final String alumniUsername;
  final String alumniEmail;
  final String professorId;
  final String professorEmail;
  final String status;
  final String createdAt;

  AlumniRequest({
    required this.id,
    required this.alumniId,
    required this.alumniUsername,
    required this.alumniEmail,
    required this.professorId,
    required this.professorEmail,
    required this.status,
    required this.createdAt,
  });

  factory AlumniRequest.fromJson(Map<String, dynamic> json) => _$AlumniRequestFromJson(json);
  Map<String, dynamic> toJson() => _$AlumniRequestToJson(this);
}